File: src/PHP/BehavioralPatterns/State/VendingMachine/SoldOutState.php
<?php

namespace BehavioralPatterns\State\VendingMachine;

class SoldOutState extends State {
    public function __construct(VendingMachine $vendingMachine) {
        parent::__construct($vendingMachine);
        echo "SOLDOUT\n";
    }

    public function insertMoney(float $amount): void {
        echo "There are no products in the vending machine.\n";
    }

    public function selectProduct(string $productCode): void {
        echo "There are no products in the vending machine.\n";
    }

    public function dispenseProduct(): void {
        echo "There is no selected product for dispensing\n";
    }

    public function cancel(): void {
        echo "There is no operation to be cancelled\n";
    }

    public function refill(array $products): void {
        $this->vendingMachine->setProducts($products);

        echo "Total amount of products: " . array_reduce($products, function ($sum, $product) {
                return $sum + $product->getStock();
            }, 0) . "\n";

        $this->vendingMachine->setState(new IdleState($this->vendingMachine));
    }
}

File: src/Java/BehavioralPatterns/State/VendingMachine/src/SoldOutState.java
import java.util.List;

public class SoldOutState extends State {
    public SoldOutState(VendingMachine vendingMachine) {
        super(vendingMachine);
        System.out.println("SOLDOUT");
    }

    @Override
    public void insertMoney(double amount) {
        System.out.println("There are no products in the vending machine.");
    }

    @Override
    public void selectProduct(String productCode) {
        System.out.println("There are no products in the vending machine.");
    }

    @Override
    public void dispenseProduct() {
        System.out.println("There is no selected product for dispensing");
    }

    @Override
    public void cancel() {
        System.out.println("There is no operation to be cancelled");
    }

    @Override
    public void refill(List<Product> products) {
        vendingMachine.setProducts(products);
        int totalStock = vendingMachine.getProducts().stream().mapToInt(Product::getStock).sum();
        System.out.println("Total amount of products: " + totalStock);
        vendingMachine.setState(new IdleState(vendingMachine));
    }
}
